src/index/IndexImpl.Text.cpp

```diff
@@ -986,10 +993,11 @@ void IndexImpl::getFilteredECListForWords(const string& words,
       }
       it->second.push_back(filter[i]);
     }
-    vector<TextRecordIndex> cids;
-    vector<Id> eids;
-    vector<Score> scores;
-    getContextEntityScoreListsForWords(words, cids, eids, scores);
+    Index::WordEntityPostings wep;
+    wep = getContextEntityScoreListsForWords(words);
+    vector<TextRecordIndex> cids = wep.cids;  // TODO: rewrite
+    vector<Id> eids = wep.eids;
+    vector<Score> scores = wep.scores;
```

Comment:
Yes, I think you should also rewrite the `oneVarFilterAggScores...` functions to return `WordEntityPostings`.